### PR TITLE
Add new board Raytac MDBT50Q-DB-40

### PR DESCRIPTION
**Pull request by Arudinne**
This is the board: https://www.raytac.com/product/ins.php?index_id=81. This board appears to be a simple breakout / demo board for all of the pins on the Raytac MDBT50Q-1MV2 module used in other products and available here: https://www.adafruit.com/product/4078

I used the nRF Connect suite and a JLINK EDU Mini to load the same UF2 bootloader used by the PCA10059 and then I was able to install CircuitPython without issue.

This is the output from boot_out.txt: 

**Adafruit CircuitPython 6.0.0-alpha.1-17-ga7844205b on 2020-07-07; MDBT50Q-DB-40 with nRF52840**

REPL via USB serial seems to work, as does copying over files via USB but I haven't had a chance to test any higher level functions.

### CODE REVIEW

File: ports/nrf/boards/raytac_mdbt50q-db-40/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,8 @@
+USB_VID = 0x239A
+USB_PID = 0x802A
```
Comment (tannewt) [inline ports/nrf/boards/raytac_mdbt50q-db-40/mpconfigboard.mk:2]: Please don't reuse the PID. See the learn guide for requesting a new PID.

File: ports/nrf/boards/raytac_mdbt50q-db-40/pins.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    { MP_ROM_QSTR(MP_QSTR_NFC_2), MP_ROM_PTR(&pin_P0_10) },
+    
+    { MP_ROM_QSTR(MP_QSTR_TX), MP_ROM_PTR(&pin_P0_06) },
+    { MP_ROM_QSTR(MP_QSTR_RX), MP_ROM_PTR(&pin_P0_08) },
+    { MP_ROM_QSTR(MP_QSTR_CTS), MP_ROM_PTR(&pin_P0_07) },
+    { MP_ROM_QSTR(MP_QSTR_RTS), MP_ROM_PTR(&pin_P0_05) },
+    
+    { MP_ROM_QSTR(MP_QSTR_LED_R), MP_ROM_PTR(&pin_P0_14) },
+    { MP_ROM_QSTR(MP_QSTR_LED_G), MP_ROM_PTR(&pin_P0_13) },
+    { MP_ROM_QSTR(MP_QSTR_LED_B), MP_ROM_PTR(&pin_P0_15) },
```
Comment (tannewt) [inline ports/nrf/boards/raytac_mdbt50q-db-40/pins.c:71]: We do support a tricolor as a status NeoPixel. See here for an example: https://github.com/adafruit/circuitpython/blob/main/ports/nrf/boards/particle_argon/mpconfigboard.h#L35

Comment (Arudinne) [reply ports/nrf/boards/raytac_mdbt50q-db-40/pins.c:71]: This board doesn't have a neopixel, it has 3 separate LEDs and one spot for a 4th LED of the user's choice. Or at least I assume that is Raytac's intention. It costs roughly $20 USD at the time of this post and it's the only nRF52840 board I know that breaks out all of the pins at that price point.

File: ports/nrf/boards/raytac_mdbt50q-db-40/mpconfigboard.mk (hunk 1 above)
Comment (Arudinne) [reply ports/nrf/boards/raytac_mdbt50q-db-40/mpconfigboard.mk:2]: Request submitted.

Comment (Arudinne) [reply ports/nrf/boards/raytac_mdbt50q-db-40/mpconfigboard.mk]: PID changed.

**Comment by Arudinne** [comment]
I'm sure why the merge check is failing. It seems to be indicating an issue with the white space that I can't see on my end.

**Comment by dhalbert** [comment]
> I'm sure why the merge check is failing. It seems to be indicating an issue with the white space that I can't see on my end.

`pre-commit` is complaining blank lines with spaces in them in `pins.c`. So check the blank lines between the pin lines.

Check if your editor has an option or a plug-in to trim trailing whitespace when it saves the file. I have that enabled in emacs and it saves a lot of manual fixing.

I was able to `pip3 install pre-commit`, and then run it by hand. The build log nicely describes how to do that.

**Comment by Arudinne** [comment]
Looks like it should be good now?

**Comment by dhalbert** [comment]
D1, D2, and D3 are green, red, and blue, so would you be interested in pretending to use them as a single RGB status LED, similar to the PCA10059?

**Comment by Arudinne** [comment]
The PCA10059 has one single color LED and one RGB LED, where as this has 3 physically separate LEDS and an empty set of pads for a 4th LED. If you think it's better to have it pretend to be a single RGB LED I'm not entirely against it, I just don't think it accurately reflects the hardware to treat it that way.
7eddbe6a2b3f2bd8152c1df95b17f443 vs 
nRF52840_dongle_kit_content

**Comment by dhalbert** [comment]
I'm willing to go along with your current treatment of the LEDs. @tannewt do you want to review further?

**Pull request merged**